server/src/main/java/org/elasticsearch/action/search/SearchDfsQueryThenFetchAsyncAction.java

```diff
@@ -63,7 +64,7 @@ final class SearchDfsQueryThenFetchAsyncAction extends AbstractSearchAsyncAction
             clusters
         );
         this.queryPhaseResultConsumer = queryPhaseResultConsumer;
-        SearchProgressListener progressListener = task.getProgressListener();
+        this.progressListener = task.getProgressListener();
         if (progressListener != SearchProgressListener.NOOP) {
             notifyListShards(progressListener, clusters, request.source());
         }
```

Comment:
This was copied from the SearchQueryThenFetchAsyncAction, which also has the guards, but I removed the "if" check in both places, since as you say, it should be safe.